mecanum_drive_controller/CMakeLists.txt

```diff
@@ -33,29 +37,29 @@ foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
   find_package(${Dependency} REQUIRED)
 endforeach()
 
-generate_parameter_library(mecanum_drive_controller_parameters
-                           src/mecanum_drive_controller_parameter.yaml)
+# get include dirs from control_toolbox for the custom validators
+get_target_property(TB_INCLUDE_DIRS control_toolbox::rate_limiter_parameters
+                    INTERFACE_INCLUDE_DIRECTORIES)
+generate_parameter_library(
+  mecanum_drive_controller_parameters
+  src/mecanum_drive_controller_parameter.yaml
+  ${TB_INCLUDE_DIRS}/control_toolbox/custom_validators.hpp)
 
-add_library(
-  mecanum_drive_controller SHARED src/mecanum_drive_controller.cpp
-                                  src/odometry.cpp src/speed_limiter.cpp)
+add_library(mecanum_drive_controller SHARED src/mecanum_drive_controller.cpp
+                                            src/odometry.cpp)
+target_compile_features(mecanum_drive_controller PUBLIC cxx_std_17)
 target_include_directories(
   mecanum_drive_controller
-  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/mecanum_drive_controller>)
-target_link_libraries(mecanum_drive_controller
-                      PUBLIC mecanum_drive_controller_parameters)
+target_link_libraries(
+  mecanum_drive_controller PUBLIC mecanum_drive_controller_parameters
+                                  control_toolbox::rate_limiter_parameters)
 ament_target_dependencies(mecanum_drive_controller PUBLIC
                           ${THIS_PACKAGE_INCLUDE_DEPENDS})
-# Causes the visibility macros to use dllexport rather than dllimport, which is
-# appropriate when building the dll but not consuming it.
-target_compile_definitions(mecanum_drive_controller
-                           PRIVATE "MECANUM_DRIVE_CONTROLLER_BUILDING_DLL")
 pluginlib_export_plugin_description_file(controller_interface
                                          mecanum_drive_plugin.xml)
 
-install(DIRECTORY include/ DESTINATION include/mecanum_drive_controller)
-
```

Comment:
in diff there is: https://github.com/ros-controls/ros2_controllers/blob/b75fe9eed17d485e388c5cf64f5a6c0d630912f7/diff_drive_controller/CMakeLists.txt#L85